CMake/cdat_modules_extra/git_clone.sh.in

```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
 
-cd @CMAKE_INSTALL_PREFIX@
-"@GIT_EXECUTABLE@" clone --depth 1 -b @BRANCH@ @GIT_URL@ @GIT_TARGET@
+cd "@CMAKE_INSTALL_PREFIX@"
+"@GIT_EXECUTABLE@" clone --no-checkout --depth 1 -b @BRANCH@ @GIT_URL@ "@GIT_TARGET@"
+cd "@GIT_TARGET@"
+if [ "$("@GIT_EXECUTABLE@" cat-file -t @BRANCH@)" = tag ]; then
+    "@GIT_EXECUTABLE@" checkout @BRANCH@
+else
```

Comment:
This will checkout in "detached head" mode anyway, thus not changing local branches, unfortunately "clone -b" will have already created a local branch if `@BRANCH@` is a branch name. I don't know how to stop that.